src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FTCategoryUpdateServiceImplTest.java

```diff
@@ -0,0 +1,89 @@
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryUpdateDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import java.util.Date;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class FTCategoryUpdateServiceImplTest {
+    private static final long VALID_ID = 1L;
+    private static final long INVALID_ID = 10L;
+
+    private static final String NAME = "test_name";
+    private static final FinancialTransactionType TYPE = FinancialTransactionType.INCOME;
+
+    private static final FinancialTransactionCategory VALID_CATEGORY
+            = FinancialTransactionCategory.builder()
+            .id(VALID_ID)
+            .name(NAME)
+            .type(TYPE)
+            .creation_date(new Date())
+            .build();
+
+    private static final FinancialTransactionCategoryDTO VALID_CATEGORY_DTO
+            = new FinancialTransactionCategoryDTO(VALID_ID,NAME,TYPE);
+    private static final FinancialTransactionCategoryUpdateDTO VALID_UPDATE_CATEGORY_DTO
+            = new FinancialTransactionCategoryUpdateDTO(NAME,TYPE);
+
+    @Mock
+    private FinancialTransactionCategoryRepository repository;
+
+    @Mock
+    private FinancialTransactionCategoryModelMapper mapper;
+
+
+    @InjectMocks
+    private FinancialTransactionCategoryServiceImpl service;
+
+    @BeforeEach
+    void setUp(){
+        Mockito.when(repository.findById(VALID_ID)).thenReturn(Optional.of(VALID_CATEGORY));
+        Mockito.when(repository.findById(INVALID_ID)).thenReturn(Optional.empty());
+        Mockito.when(repository.save(VALID_CATEGORY)).thenReturn(VALID_CATEGORY);
+        Mockito.when(mapper.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(ArgumentMatchers.any())).thenReturn(VALID_CATEGORY_DTO);
+    }
+
+    @Test
+    @DisplayName("update FT Category if id is correct and return the category object")
+    void testUpdateFTCategoryById_WhenIdIsCorrect_ThenReturnCategoryEntity(){
+        FinancialTransactionCategoryDTO entity
+                = service.updateFinancialTransactionCategory(VALID_ID,VALID_UPDATE_CATEGORY_DTO);
+        Assertions.assertEquals(entity,VALID_CATEGORY_DTO);
+        Mockito.verify(repository,Mockito.times(1)).save(VALID_CATEGORY);
```

Comment:
Instead of:
`Mockito.verify(repository,Mockito.times(1)).save(VALID_CATEGORY);`
maybe
`Mockito.verify(repository, Mockito.times(1)).save(VALID_CATEGORY);`
?